src/parser.js

```diff
@@ -0,0 +1,782 @@
+// The main Swagger parsing component that outputs refract.
+
+import _ from 'lodash';
+import Ast from './ast';
+import buildUriTemplate from './uri-template';
+import SwaggerParser from 'swagger-parser';
+import yaml from 'js-yaml';
+
+// These describe the type of annotations that are produced by this parser
+// and assigns a unique code to each one. Downstream applications can use this
+// code to group similar types of annotations together.
+const ANNOTATIONS = {
+  CANNOT_PARSE: {
+    type: 'error',
+    code: 1,
+    fragment: 'yaml-parser',
+  },
+  AST_UNAVAILABLE: {
+    type: 'warning',
+    code: 2,
+    fragment: 'yaml-parser',
+  },
+  DATA_LOST: {
+    type: 'warning',
+    code: 3,
+    fragment: 'refract-not-supported',
+  },
+  VALIDATION_ERROR: {
+    type: 'warning',
+    code: 4,
+    fragment: 'swagger-validation',
+  },
+};
+
+// Test whether a key is a special Swagger extension.
+function isExtension(value, key) {
+  return key.indexOf('x-') === 0;
+}
+
+// The parser holds state about the current parsing environment and converts
+// the input Swagger into Refract elements. The `parse` function is its main
+// interface.
+class Parser {
+  constructor({minim, source, generateSourceMap}) {
+    // Parser options
+    this.minim = minim;
+    this.source = source;
+    this.generateSourceMap = generateSourceMap;
+
+    // Loaded, dereferenced Swagger API
+    this.swagger = null;
+    // Refract parse result
+    this.result = null;
+    // Refract API category
+    this.api = null;
+    // State of the current parsing path
+    this.path = [];
+    // Current resource group, if any
+    this.group = null;
+  }
+
+  parse(done) {
+    const {
+      Copy, Category, Member: MemberElement, ParseResult, SourceMap,
+    } = this.minim.elements;
+
+    const parser = new SwaggerParser();
```

Comment:
Just wondering if you can call it something swagger-specific to not overload the term "parse" here.